browser/ui/webui/sanitized_image_source_unittest.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "chrome/browser/ui/webui/sanitized_image_source.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/files/file_util.h"
+#include "base/memory/ref_counted_memory.h"
+#include "base/path_service.h"
+#include "base/test/bind.h"
+#include "brave/components/constants/brave_paths.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/test/base/chrome_test_utils.h"
+#include "chrome/test/base/testing_profile.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/test/browser_task_environment.h"
+#include "services/data_decoder/public/cpp/test_support/in_process_data_decoder.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/cpp/weak_wrapper_shared_url_loader_factory.h"
+#include "services/network/test/test_url_loader_factory.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace {
+constexpr char kValidImage[] = "ad_banner.png";
+constexpr char kPaddedImage[] = "padded.png.pad";
+constexpr char kInvalidImage[] = "normal.js";
+
+const std::string LoadTestFile(const std::string& name) {
```

Comment:
nit: pls fix return value `const std::string` -> `std::string`. Currently you're most likely prevent r-value optimization and do an unnecessary copy here: `std::string data = LoadTestFile(filename);`.